src/parser/core/modules/Gauge/CounterGauge.ts

```diff
@@ -50,8 +69,10 @@ export class CounterGauge extends AbstractGauge {
 		super(opts)
 
 		this.minimum = opts.minimum || 0
-		this._value = opts.initialValue || this.minimum
+		this.initialValue = opts.initialValue || this.minimum
+		this._value = this.initialValue
 		this.maximum = opts.maximum || 100
+		this.correctHistory = opts.correctHistory || false
```

Comment:
Especially now that `false` is a real default value for one of these options, we should use ?? coalescing rather than falsey to determine that we need to use the default.